File: bongf/java/src/ch12_implementation_qeustions/Q8_StringResort.java
package ch12_implementation_qeustions;

import java.util.*;
import java.util.stream.Collectors;

public class Q8_StringResort {
    public static void main(String[] args) {
        Q8_StringResort sr = new Q8_StringResort();
//        System.out.println(sr.solution());
//        System.out.println(sr.dongbin());
    }

    public String solution() {
        Scanner sc = new Scanner(System.in);
        char[] chars = sc.nextLine().toCharArray();
        List<Character> r = new ArrayList<>();
        int s = 0;
        int count = 0;
        for ( char c : chars) {
            if(  'A' <= c && c <= 'Z') {
                r.add(c);
            } else {
                s += Integer.parseInt(c+"");
                count++;
            }
        }

        Collections.sort(r);
        String collect = r.stream().map(String::valueOf).collect(Collectors.joining());
        return count == 0 ? collect : collect + String.valueOf(s);
    }

    // 애초에 배열을 안만들고 characterAt을 사용
    // Character.isLetter 메소드 사용
    public String dongbin() {
        Scanner sc = new Scanner(System.in);
        String str = sc.next();
        List<Character> result = new ArrayList<>();

        int value = 0;
        int count = 0;

        for (int i = 0; i < str.length(); i++) {
            if(Character.isLetter(str.charAt(i))) {
                result.add(str.charAt(i));
            } else {
                value += str.charAt(i) - '0';
                count++;
            }
        }

        Collections.sort(result);
        String collect = result.stream().map(String::valueOf).collect(Collectors.joining());
        return count == 0 ? collect : collect + String.valueOf(value);
    }
}

/*
K1KA5CB7
결과 ABCKK13
AJKDLSI412K4JSJ9D
결과 ADDIJJJKKLSS20
 */

File: bongf/python/dongbinbook/ch12_implementation_questions/q8_string_resort.py
def solution() :
    n = list(input())
    n.sort()
    sum = 0 
    count = 0 
    while True :
        s = n[0]
        if not s.isdigit() :
            break
        sum += int(n.pop(0))
        count += 1
    
    return "".join(n) + str(sum) if count != 0 else "".join(n)

print(solution())

## 동빈북 풀이보면 숫자가 0이 들어올 경우는 0출력 안해주는 걸로 하는데 그냥 해주는 걸로 바꿨다 
def dongbin() :
    data = input()
    result = [] 
    value = 0 
    count = 0 

    for x in data :
        if x.isalpha() :
            result.append(x)
        else :
            value += int(x)
            count += 1
    
    result.sort()

    if count == 0 :
        return "".join(result) 
    return "".join(result) + str(sum)
    
'''
K1KA5CB7
결과 ABCKK13
AJKDLSI412K4JSJ9D
결과 ADDIJJJKKLSS20
'''
